src/Microsoft.Extensions.FileProviders.Sources/EnumerableDirectoryContents.cs

```diff
@@ -22,12 +41,19 @@ public bool Exists
 
         public IEnumerator<IFileInfo> GetEnumerator()
         {
+            EnsureInitialized();
             return _entries.GetEnumerator();
         }
 
         IEnumerator IEnumerable.GetEnumerator()
         {
+            EnsureInitialized();
             return _entries.GetEnumerator();
         }
+
+        private void EnsureInitialized()
+        {
+            LazyInitializer.EnsureInitialized(ref _entries, _entriesFactory);
```

Comment:
Probably don't need any of the behavior of `LazyInitializer`. if (_entries == null) { .. }` should suffice.